internal/operands/tekton-tasks/tekton-tasks_test.go

```diff
@@ -27,106 +31,190 @@ const (
 			}
-		}
-	})
+		})
 
-	It("Name function should return correct name", func() {
-		name := tt.Name()
-		Expect(name).To(Equal(operandName), "should return correct name")
-	})
+		It("should remove tekton-tasks resources on cleanup", func() {
+			_, err := operand.Reconcile(&request)
+			Expect(err).ToNot(HaveOccurred())
 
-	It("Reconcile function should return correct functions", func() {
-		functions, err := tt.Reconcile(mockedRequest)
-		Expect(err).ToNot(HaveOccurred(), "should not throw err")
-		Expect(len(functions)).To(Equal(8), "should return correct number of reconcile functions")
-	})
+			for _, task := range bundle.Tasks {
+				ExpectResourceExists(&task, request)
+			}
+
+			for _, clusterRole := range bundle.ClusterRoles {
+				ExpectResourceExists(&clusterRole, request)
+			}
+
+			for _, serviceAccount := range bundle.ServiceAccounts {
+				ExpectResourceExists(&serviceAccount, request)
+			}
+
+			for _, roleBinding := range bundle.RoleBindings {
+				ExpectResourceExists(&roleBinding, request)
+			}
+
+			_, err = operand.Cleanup(&request)
+			Expect(err).ToNot(HaveOccurred())
 
-	It("RequiredCrds function should return required crds", func() {
-		tt := getMockedTektonTasksOperand()
-		crds := tt.RequiredCrds()
+			for _, task := range bundle.Tasks {
+				ExpectResourceNotExists(&task, request)
+			}
+
+			for _, clusterRole := range bundle.ClusterRoles {
+				ExpectResourceNotExists(&clusterRole, request)
+			}
 
-		Expect(len(crds) > 0).To(BeTrue(), "should return required crds")
+			for _, serviceAccount := range bundle.ServiceAccounts {
+				ExpectResourceNotExists(&serviceAccount, request)
+			}
 
-		for _, crd := range crds {
-			found := false
-			for _, c := range requiredCRDs {
-				if crd == c {
-					found = true
-				}
+			for _, roleBinding := range bundle.RoleBindings {
+				ExpectResourceNotExists(&roleBinding, request)
 			}
-			Expect(found).To(BeTrue(), "should return correct required crd")
-		}
+		})
 	})
 
+	Context("With feature gate disabled", func() {
+		BeforeEach(func() {
+			request.Instance.Spec.FeatureGates.DeployTektonTaskResources = false
+		})
+
+		It("Should not create tekton-tasks resources", func() {
+			_, err := operand.Reconcile(&request)
+			Expect(err).ToNot(HaveOccurred())
+
+			for _, task := range bundle.Tasks {
+				ExpectResourceNotExists(&task, request)
+			}
+
+			for _, clusterRole := range bundle.ClusterRoles {
+				ExpectResourceNotExists(&clusterRole, request)
+			}
+
+			for _, serviceAccount := range bundle.ServiceAccounts {
+				ExpectResourceNotExists(&serviceAccount, request)
+			}
+
+			for _, roleBinding := range bundle.RoleBindings {
+				ExpectResourceNotExists(&roleBinding, request)
+			}
+		})
+	})
 })
 
-func TestTektonBundle(t *testing.T) {
+func TestTektonTasks(t *testing.T) {
 	RegisterFailHandler(Fail)
-	RunSpecs(t, "tekton tasks Suite")
+	RunSpecs(t, "Tekton Tasks Suite")
 }
 
-func getMockedRequest() *common.Request {
+func getMockedRequest() common.Request {
```

Comment:
Can you return pointer instead of a copy?

Reply:
Done.